src/target_set.c

```diff
@@ -238,17 +421,75 @@ static int ts_invoke (const data_set_t *ds, value_list_t *vl, /* {{{ */
     return (-EINVAL);
   }
 
+  orig = *vl;
+
   if (data->meta != NULL)
   {
-    meta_data_clone_merge(&(vl->meta), data->meta);
+    char temp[DATA_MAX_NAME_LEN*2];
+    int meta_entries;
+    char **meta_toc;
+
+    if ((new_meta = meta_data_create()) == NULL)
+    {
+      ERROR ("Target `set': failed to create replacement metadata.");
+      return (-ENOMEM);
+    }
+
+    meta_entries = meta_data_toc (data->meta, &meta_toc);
+    for (int i = 0; i < meta_entries; i++)
+    {
+      const char *key = meta_toc[i];
+      char *string;
+      int status;
+
+      status = meta_data_get_string (data->meta, key, &string);
```

Comment:
This will result in two errors (one in `meta_data_get_string()` and one here) for each non-string meta data entry.

You have a long-ish `switch (meta_type)` above. How about we move this into `src/daemon/meta_data.c` as `meta_data_as_string()` which returns a meta data entry as string, regardless of it's actual type. We could then re-use that here.